Neo4j.Driver/Neo4j.Driver/Internal/ConnectionValidator.cs

```diff
@@ -0,0 +1,116 @@
+using System;
+using System.Threading.Tasks;
+
+namespace Neo4j.Driver.Internal
+{
+    internal class ConnectionValidator
+    {
+        private readonly TimeSpan _connIdleTimeout;
+        private readonly TimeSpan _maxConnLifeTime;
+
+        public ConnectionValidator(TimeSpan connIdleTimeout, TimeSpan maxConnLifeTime)
+        {
+            _connIdleTimeout = connIdleTimeout;
+            _maxConnLifeTime = maxConnLifeTime;
+        }
+
+        public bool IsConnectionReusable(IPooledConnection connection)
+        {
+            if (!connection.IsOpen)
+            {
+                return false;
+            }
+
+            try
+            {
+                connection.ClearConnection();
```

Comment:
Not specific for this PR, but if in the middle of a huge resultset streaming, calling RESET may stop the server from streaming more data but it will not help client's consuming pending network buffers - which may take considerable time. Shall we have a means of timeout-ing?